src/midi/fluid_midi.c

```diff
@@ -1604,6 +1604,10 @@ fluid_track_send_events(fluid_track_t *track,
             if(player->playback_callback)
             {
                 player->playback_callback(player->playback_userdata, event);
+                if(event->type == NOTE_ON && !player->channel_isplaying[event->channel])
```

Comment:
I think we should also send ALL_CTRL_OFF in some cases, so in my opinion we don't need to filter on the event type here. So any MIDI message on a channel marks that channel as `isplaying`.

Reply:
Like I said in the discussion, I would be fine if we refrain from sending ALL_CRTL_OFF for now. Checking for the event type is ok for me. However, you also have to check for `velocity != 0` to be sure it's a real noteon.

Reply:
I added checking for note velocity as well - I like limiting the channels that receive SOUND_OFF to the minimum number necessary